tests/test_coordinated_workers/test_coordinator.py

```diff
@@ -344,3 +347,72 @@ def test_invalid_databag_content(coordinator_charm: ops.CharmBase, event):
         cluster.gather_addresses_by_role()
         manager.run()
     assert cluster.model.unit.status == ops.BlockedStatus("[consistency] Cluster inconsistent.")
+
+
+@patch('cosl.coordinated_workers.interface.ClusterProviderAppData.load',
+       side_effect=DataValidationError("Mock error"))
+def test_invalid_coordinator_databag(mock_databag_load, coordinator_charm: ops.CharmBase, coordinator_state):
+    # Test that when a relation changes and the relation app data is invalid
+    #   the worker emits a ClusterRemovedEvent
+
+    # WHEN you define a properly configured charm
+    ctx = testing.Context(
+        MyCharm,
+        meta={
+            "name": "foo",
+            "requires": {"cluster": {"interface": "cluster"}},
+            "containers": {"foo": {"type": "oci-image"}},
+        },
+        config={"options": {"role-all": {"type": "boolean", "default": True}}},
+    )
+
+    # AND the relation changes
+    # IF the relation data is invalid (forced by the patched Exception)
+    relation = testing.Relation(
+        "cluster",
+        remote_app_data={"worker_config": json.dumps("some: yaml")},
+    )
+
+    # THEN the charm emits a ClusterRemovedEvent
+    with ctx(ctx.on.relation_changed(relation),
```

Comment:
since you're not using the manager at all, this should be replaced with the regular `ctx.run()` flow